### PR TITLE
Fix starts_with fields on `PresignedPost`

### LINKED ISSUE #2636: Unsupported option: foo_starts_with after upgrading aws-sdk-s3 from 1.111.1 to 1.111.2
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've gone though [Developer Guide for v3](https://docs.aws.amazon.com/sdk-for-ruby/v3/developer-guide/welcome.html) and [API reference](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/index.html)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-ruby) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-ruby/issues) and didn't find any solution
  
**Describe the bug**

After upgrading aws-sdk-s3 from 1.111.1 to 1.111.2 we got ArgumentErrors saying "Unsupported option: content_type_starts_with". Our code effectively does

    Aws::S3::Resource.new.bucket("the-name").presigned_post({ …, content_type_starts_with: "image/" })

This worked fine before. I see the diff for this upgrade touches on "starts_with", so it must have introduced a regression.

**Gem name ('aws-sdk', 'aws-sdk-resources' or service gems like 'aws-sdk-s3') and its version**

aws-sdk-s3 1.111.2

**Version of Ruby, OS environment**
-paste the output of ruby -v

ruby 2.7.5p203 (2021-11-24 revision f69aeb8314) [x86_64-linux]

### PR DESCRIPTION
**Pull request by alextwoods**
Fixes #2636 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

1. To make sure we include your contribution in the release notes, please make sure to add description entry for your changes in the "unreleased changes" section of the `CHANGELOG.md` file (at corresponding gem). For the description entry, please make sure it lives in one line and starts with `Feature` or `Issue` in the correct format.

2. For generated code changes, please checkout below instructions first:
  https://github.com/aws/aws-sdk-ruby/blob/version-3/CONTRIBUTING.md

Thank you for your contribution!

### CODE REVIEW

File: gems/aws-sdk-core/lib/aws-sdk-core/plugins/response_paging.rb [hunk 1]
```diff
@@ -10,7 +10,9 @@ class Handler < Seahorse::Client::Handler
         def call(context)
           context[:original_params] = context.params
           resp = @handler.call(context)
+          puts RubyVM.stat[:global_constant_state]
```
Comment (mullermp) [inline gems/aws-sdk-core/lib/aws-sdk-core/plugins/response_paging.rb]: Remove

**Review by mullermp — COMMENTED** [review]
Are there any other instances where we need changes?

**Comment by mullermp** [comment]
LGTM assuming no other prefixes or suffixes or w/e

**Comment by alextwoods** [comment]
`starts_with` is the only supported option on `define_fields` so should be the only one.

**Pull request merged**